AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_script_canvas.py

```diff
@@ -38,12 +37,10 @@ class ScriptCanvasComponent:
 
     def __init__(self):
         self.editor_entity = None
-        self.script_canvas_component = None
+        self.script_canvas_components = []
```

Comment:
The script canvas component should only handle a single instance of the component. Overloading it like this puts some awkward behavior into the class that belongs to the editor entity class and not the component class. If somebody wants to add multiple script canvas components, it should not be be the the responsibility of the component class to care and should only have a reference to itself.

Please refactor this class to be closer to` editor_physx_collider.`

Reply:
component class now only manages a single script canvas component reference.